components/Common/Search/States/WithSearchResult.tsx

```diff
@@ -1,32 +1,43 @@
 import type { Result } from '@orama/orama';
-import type { FC } from 'react';
+import { useEffect, type FC, useRef } from 'react';
 
 import { pathToBreadcrumbs } from '@/components/Common/Search/utils';
 import Link from '@/components/Link';
-import { BASE_URL } from '@/next.constants.mjs';
 import { highlighter } from '@/next.orama.mjs';
 import type { SearchDoc } from '@/types';
+import { searchHitToLinkPath } from '@/util/search';
 
 import styles from './index.module.css';
 
 type SearchResultProps = {
   hit: Result<SearchDoc>;
   searchTerm: string;
+  selected: boolean;
+  idx: number;
 };
 
 export const WithSearchResult: FC<SearchResultProps> = props => {
-  const isAPIResult = props.hit.document.siteSection.toLowerCase() === 'docs';
-  const basePath = isAPIResult ? BASE_URL : '';
-  const path = `${basePath}/${props.hit.document.path}`;
+  const divRef = useRef<HTMLDivElement>(null);
+  const path = searchHitToLinkPath(props.hit);
+
+  useEffect(() => {
+    if (props.selected && divRef.current) {
+      divRef.current?.scrollIntoView({ block: 'center' });
```

Comment:
```suggestion
      divRef.current.scrollIntoView({ block: 'center' });
```

You're already checking if `divRef.current` is defined on the if statement

Reply:
True, changed it